static/js/chart/draw_bar.ts

```diff
@@ -496,16 +699,25 @@ function drawHorizontalStackedBars(
       .append("g")
       .attr("fill", (d) => colorFn(d.key))
       .selectAll("circle")
-      .data((d) => d.map((dp) => ({ key: d.key, ...dp })))
+      .data((d) =>
+        d.map((dp) => ({
```

Comment:
nit: it looks like this same mapping is done multiple times in this file.. could we make a helper function for this?